core/src/stages/move_relative.cpp

```diff
@@ -104,6 +104,63 @@ static bool getJointStateFromOffset(const boost::any& direction, const moveit::c
 	return false;
 }
 
+static void visualizePlan(Interface::Direction dir, bool success, std::deque<visualization_msgs::Marker>& markers,
+                          const std::string& ns, const std::string& frame_id, const Eigen::Isometry3d& link_pose,
+                          const Eigen::Isometry3d& reached_pose, const Eigen::Vector3d& linear, double distance) {
+	auto linear_norm = linear.norm();
+
+	// rotation of the target direction and for the cylidner marker
+	auto quat_target = Eigen::Quaterniond::FromTwoVectors(Eigen::Vector3d::UnitX(), linear);
+	auto quat_cylinder = quat_target * Eigen::AngleAxisd(0.5 * M_PI, Eigen::Vector3d::UnitY());
+
+	// link position before planning; reached link position after planning; target link position
+	Eigen::Vector3d pos_link = link_pose.translation();
+	Eigen::Vector3d pos_reached = reached_pose.translation();
+	Eigen::Vector3d pos_target = pos_reached + quat_target * Eigen::Vector3d(linear_norm - distance, 0, 0);
```

Comment:
```suggestion
	auto& pos_link = link_pose.translation();
	auto& pos_reached = reached_pose.translation();
	auto pos_target = pos_reached + quat_target * Eigen::Vector3d(linear_norm - distance, 0, 0);
```
... is what I would write, because the explicit types will actually evaluate Eigen's templated lazy evaluation before it's actually needed (the auto-deduced types are not Vector3d, but some `EigenBase<...>` datatypes). It won't make a difference here though, so feel free to leave it as is. See for example [here](https://eigen.tuxfamily.org/dox/TopicFunctionTakingEigenTypes.html) for something to start reading about the subject.

Reply:
I am unsure why this doesn't work, but I tried it with `auto` and `auto&` before and just did again and it breaks the visualization. Since I simply pass the variables to `tf::pointEigenToMsg`, it has to be something in there, but I didn't do any further troubleshooting.